### PR TITLE
feat(chat): regenerate redesign (Issue #878)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

Move regenerate to the last message

Issues:

- https://github.com/epam/ai-dial-chat/issues/878

**UI changes**

https://www.figma.com/file/lwL5jneveZczAj5RLcLFH2/DIAL-Chat?type=design&node-id=7001-10995&mode=design&t=1AJfx1hYukByRz95-4

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-897.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Chat/Chat.tsx [hunk 1]
```diff
@@ -725,35 +741,50 @@ export const ChatView = memo(() => {
...
-                                    )}
-                                    editDisabled={!!notAllowedType}
-                                    onEdit={onEditMessage}
-                                    onLike={onLikeHandler(index, conv)}
-                                    onDelete={() => {
-                                      handleDeleteMessage(index);
-                                    }}
-                                  />
+                            ]) => {
+                              if (
```
Comment (mikitabut) [inline apps/chat/src/components/Chat/Chat.tsx]: I think it's bad practise to set any usual logic inside the rendered HTML like changing state

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-897.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-897.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**